soccer/gameplay/plays/testing/four_courners_pass.py

```diff
@@ -0,0 +1,229 @@
+		# decide which direction you're passing to
+		self.safer_pass()
+		# based on which direction you're passing to pick that index
+		kickToIdx = (kickFromIdx + self.direction) % len(self.square_points)
+		# get that point
+		kickToPt = self.square_points[kickToIdx]
+
+		# remove the subbehavior that's not recieving the pass
+		if (kickToPt is self.otherPts[0]) :
+			self.remove_subbehavior('move1')
+			self.text = 'move2'
+		else :
+			self.remove_subbehavior('move2')
+			self.text = 'move1'
+
+		# while pass is preparing get ready for the next pass
+		premove = skills.move.Move(self.square_points[(kickToIdx + self.direction) % len(self.square_points)])
+		# get ready to pass the ball
+		self.passing_machine = tactics.coordinated_pass.CoordinatedPass(kickToPt)
+
+		#if (self.has_subbehavior_with_name('Chasing')) :
+		#	self.remove_subbehavior('Chasing')
+
+		self.add_subbehavior(self.passing_machine, 'passer', required = True, priority = 10)
+		self.add_subbehavior(premove, 'premove', required = True)
+
+
+		# add a robot to chase after the ball.
+		if (not self.has_subbehavior_with_name('Chasing')):
+			self.chaser = skills.move.Move(self.cut_of_pos(main.ball().pos))
+			self.add_subbehavior(self.chaser, 'Chasing', required = True, priority = 2)
+
+		# if we have too many robots isolate one of the robots so they don't help in the play
+		if (not self.has_subbehavior_with_name('Iso') and len(main.our_robots()) == 6) :
+			self.iso = skills.move.Move(robocup.Point(-constants.Field.Width/2, 0))
+			self.add_subbehavior(self.iso,'Iso', required = True, priority = 1)
+			
+	# remove subbehaviors 
+	def on_exit_passing(self):
+		self.remove_subbehavior(self.text)
+		self.remove_subbehavior('passer')
+		self.remove_subbehavior('Chasing')
+		self.remove_subbehavior('premove')
+
+	# takes in the square points and form lines and create a square on the field
+	def paint_the_field(self):
+		for i in range(len(self.square_points)) :
+			main.system_state().draw_line(robocup.Line(self.square_points[i], self.square_points[(i + 1) % 4]), (135, 0, 255), "Square")
+
+	#TODO decide which the better direction to pass is
+	def safer_pass(self):
+		if (True):
+			self.direction = 1
+		else :
+			self.direction = -1
+
+	# checks if capture is completed and setup is finish
+	def is_setup_done(self):
+		return (self.subbehavior_with_name('capture').state == behavior.Behavior.
+			State.completed)
+
+	# if a point is outside of a smaller box that is the square points - 3 robot radius'
+	# return the closest point inside that box.
+	def cut_of_pos(self, point):
+		x = point.x
+		if (point.x > (self.max_x - 3 * self.bot)):
+			x = self.max_x - 3 * self.bot
+		elif (point.x < (self.min_x + 3 *self.bot)):
+			x = self.min_x + 3 *self.bot
+
+		y = point.y
+		if (point.y > (self.max_y - 3 * self.bot)):
+			y = self.max_y - 3 * self.bot
+		elif(point.y < (self.min_y + 3 * self.bot)):
+			y = self.min_y + 3 * self.bot
+		return robocup.Point(x,y)
```

Comment:
If we wait for #1236 to get merged, you can do this in two lines using the `pad` function in line with the `intersects` function

Reply:
Oh wait, this could be done even easier. This is the same thing you are doing here, but the computation is in other places. (I haven't tested this, but it should work)

```
# Projects the point given onto the edge of the smaller rectangle
# Each coordinate is found independently so we don't have to project to each
# segment of the rectangle and choose the closest
segment_x = robocup.Segment(robocup.Point(self.min_x + 3*self.bot, 0),
                            robocup.Point(self.max_x - 3*self.bot, 0))
segment_y = robocup.Segment(robocup.Point(0, self.min_y + 3*self.bot),
                            robocup.Point(0, self.max_y - 3*self.bot))

x_pos = segment_x.nearest_point(point)
y_pos = segment_y.nearest_point(point)

return robocup.Point(x_pos.x(), y_pos.y())
```

Reply:
fixed